### ISSUE TITLE
[Sol-Yul] Gnosis MultiSigWallet fails with storage update of array

### ISSUE THREAD

**Issue by axic** [ethereum/solidity#9714]
I have a translated branch here: https://github.com/axic/MultiSigWallet/tree/solidity-0.7.x

This is the failure:
```
$ solc/solc --ir MultiSigWallet/contracts/MultiSigWallet.sol 
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/YulUtilFunctions.cpp(1493): Throw in function auto solidity::frontend::YulUtilFunctions::updateStorageValueFunction(const solidity::frontend::Type &, const std::optional<unsigned int> &)::(anonymous class)::operator()() const
Dynamic exception type: boost::wrapexcept<solidity::langutil::UnimplementedFeatureError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

**Comment by chriseth** [comment]
Might be fixed by https://github.com/ethereum/solidity/pull/9602

**Comment by mijovic** [comment]
Part of #9725

**Comment by mijovic** [comment]
If it is not fixed by now (as copying of arrays, except `bytes`, to storage is implemented), it should be fixed by #10083 (implements copying of `bytes` to storage)

**Comment by mijovic** [comment]
It now fails because of `readFromStorageReferenceType()` is not implemented for arrays (#10128)

**Comment by axic** [comment]
Yep, that's correct:
```
$ solc/solc --bin MultiSigWallet/contracts/MultiSigWallet.sol --experimental-via-ir
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/YulUtilFunctions.cpp(2037): Throw in function std::__1::string solidity::frontend::YulUtilFunctions::readFromStorageReferenceType(const solidity::frontend::Type &)
Dynamic exception type: boost::wrapexcept<solidity::langutil::UnimplementedFeatureError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

So much easier with this PR...

**Comment by mijovic** [comment]
On latest develop this compiles to yul successfully

**Comment by axic** [comment]
Hm, `MultiSigWallet.sol` doesn't seem to be for me:
```
Unimplemented feature:
/Users/alex/Projects/solidity/libsolidity/codegen/YulUtilFunctions.cpp(2206): Throw in function std::__1::string solidity::frontend::YulUtilFunctions::readFromStorageReferenceType(const solidity::frontend::Type &)
Dynamic exception type: boost::wrapexcept<solidity::langutil::UnimplementedFeatureError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

**Comment by mijovic** [comment]
Just an update here, latest `develop` fails with:
```
Unimplemented feature:
/solidity/libsolidity/codegen/ir/IRGenerator.cpp(546): Throw in function auto solidity::frontend::IRGenerator::generateImplicitConstructors(const solidity::frontend::ContractDefinition &)::(anonymous class)::operator()() const
Dynamic exception type: boost::wrapexcept<solidity::langutil::UnimplementedFeatureError>
std::exception::what: Modifiers not implemented yet.
[solidity::util::tag_comment*] = Modifiers not implemented yet.
```

Rebasing on top of #10445 fails with same old error as before, than I implemented it in #10552 and than it compiles to yul
